### ISSUE TITLE
Custom function adding before augmentation in ImageDataGenerator.

### ISSUE THREAD

**Issue by omerferhatt** [tensorflow/tensorflow#44098]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests, and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.3.1
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behaviour/state.**
tensorflow.keras.preprocessing.image.ImageDataGenerator class takes an argument to implement custom function after augmentation, which it is 'preproccessing_function'. Allowing to use a custom function before augmentation or rescale/resizing will be a great advantage for images that need to be cropped into bounding boxes.

**Will this change the current API? How?**
I cannot comment on this due to my inexperience.

**Who will benefit with this feature?**
Who has bounding box coordinates in *.csv file. But not applied yet.
 
**Any Other info.**
No.

**Comment by fchollet** [comment]
ImageDataGenerator is legacy. We recommend that you do image data augmentation via layers, like this: https://keras.io/examples/vision/image_classification_from_scratch/#using-image-data-augmentation

In this setup you can add transformation layers either before or after the data augmentation layers.

**Comment by omerferhatt** [comment]
Thank you, I didn't know that ImageDataGenerator is legacy.
